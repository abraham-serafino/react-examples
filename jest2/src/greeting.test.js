import Greeting from './greeting';
import {mount} from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

describe('Greeting', () => {

  it('should match snapshot', () => {
    const tree = renderer.create(<Greeting />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('handle text change', () => {
    const name = 'Mark';

    const wrapper = mount(<Greeting />);
    const input = wrapper.find('input');
    const div = wrapper.find('.message');

    input.simulate('change', {target: {value: name}});

    expect(div.text()).toBe(`Hello, ${name}!`);
  });
});
