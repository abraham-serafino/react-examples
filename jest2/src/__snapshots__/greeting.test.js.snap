exports[`Greeting should match snapshot 1`] = `
<form>
  <div>
    <label>
      Name: 
    </label>
    <input
      onChange={[Function]}
      type="text"
      value="World" />
  </div>
  <div
    className="message">
    Hello
    , 
    World
    !
  </div>
</form>
`;
